Feature/appc 1434 automatic purchase trigger after installation wallet

brunovarelaAptoide opened the pull request:
**What does this PR do?**

-This PR removes the toast from the billing flow and when trying to buy an item with no wallet installed, the user installs the wallet and continues the purchase without never leaving it.

**Database changed?**

    No

**Where should the reviewer start?**

InstallDialogActivity.java
WalletUtils.java
BuyItemProperties.java
AppcoinsBillingStubHelper.java

**How should this be manually tested?**

Not having wallet installed and start the wallet install flow.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1434](https://aptoide.atlassian.net/browse/APPC-1434)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 
No



**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
Changes

-Toast Removed

brunovarelaAptoide committed:
changes

- buy flow after connection

brunovarelaAptoide committed:
Merge branch 'dev' into feature/APPC-1434-Automatic-purchase-trigger-after-wallet-installation

brunovarelaAptoide committed:
Changes

- Code reorganization

brunovarelaAptoide committed:
Changes

- Added a card before creating wallet adress

brunovarelaAptoide committed:
Changes

- the new card now has a centered spinner while waiting for connection to the wallet.

brunovarelaAptoide committed:
Changes

- Code cleanning
- Change instead of getting resources by id we get resources by name(Unity UDP Integration)

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -8,43 +8,39 @@
-  public AppcoinsBillingStubHelper() {
-    this.lockThread = new Object();
+  public final static String APPCOINS_BILLING_STUB_HELPER_INSTANCE = "appcoins_billing_stub_helper";
+  public final static String BUY_ITEM_PROPERTIES = "buy_item_properties";
```
Can be private

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -267,11 +188,18 @@ private boolean createRepository() {
     }
   }
 
+  public static AppcoinsBillingStubHelper getInstance() {
```
Can be package-private

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -93,60 +84,31 @@ public Bundle getSkuDetails(final int apiVersion, final String packageName, fina
+      BuyItemProperties buyItemProperties =
+          new BuyItemProperties(apiVersion, packageName, sku, type, developerPayload);
+
       Intent intent = new Intent(WalletUtils.context.get(), InstallDialogActivity.class);
```
Extract activity (context.get()) to a local variable as you're using it twice

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -187,49 +143,22 @@ private String map(SkuDetails skuDetails) {
   }
 
-  private boolean createRepository() {
+  public boolean createRepository(
```
You can return context.bindService and then remove the else. Can be package-private

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/InstallDialogActivity.java:
```diff
@@ -1,19 +1,78 @@
     WalletUtils.promptToInstallWallet();
+
+    this.
+        setContentView(this.getResources()
```
Avoid line break

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/InstallDialogActivity.java:
```diff
@@ -1,19 +1,78 @@
+            .getIdentifier(LOADING_DIALOG_CARD, "layout", this.getPackageName()));
+    loadingDialogInstall = this.findViewById(this.getResources()
+        .getIdentifier(LOADING_DIALOG_CARD, "id", this.getPackageName()));
+    loadingDialogInstall.setVisibility(View.INVISIBLE);
```
Are the "this" needed?

JDSM01 replied:
Pass this logic to onResume after the if

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -244,17 +173,9 @@ private boolean createRepository() {
-            Log.d(TAG, "onServiceConnected() called service = [" + serviceAppcoinsBilling + "]");
-          }
+          serviceAppcoinsBilling = Stub.asInterface(service);
+          connectToWalletBillingService.isConnected();
```
isConnected can be misleading as this is not a boolean method, please change to a more proper name

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java:
```diff
@@ -89,8 +88,12 @@ public static void promptToInstallWallet() {
       return;
```
Change the intentServices if logic we talked about

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java:
```diff
@@ -89,8 +88,12 @@ public static void promptToInstallWallet() {
+    dialogWalletInstall.show();
+  }
+
+  public static void dismissDialogWalletInstall(){
```
Can be package private

JDSM01 commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/wallet/DialogWalletInstall.java:
```diff
@@ -115,8 +114,8 @@ private void buildTop() {
 
```
Null check on getWindow

JDSM01 commented on android-appcoins-billing/src/main/res/layout/install_dialog_activity.xml:
```diff
@@ -3,5 +3,5 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@android:color/transparent">
+    android:background="@android:color/black">
```
Use our color reference instead of the android one

JDSM01 commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/ConnectToWalletBillingService.java:
```diff
@@ -0,0 +1,5 @@
+package com.appcoins.sdk.billing;
+
+public interface ConnectToWalletBillingService {
+  void isConnected();
```
Change name as mentioned above

brunovarelaAptoide committed:
Code review

- Code optimization
- Code cleanning

brunovarelaAptoide committed:
Code review changes

- Changed packageName initialization

brunovarelaAptoide committed:
Code review changes

- Added a response in case startActivityforResult throws a exception in InstallDialogActivity
-Removed unused colors.
